mithril-common/src/crypto_helper/merkle_tree.rs

```diff
@@ -257,10 +359,47 @@ mod tests {
         let leaves_to_verify = &[leaves[0].to_owned(), leaves[3].to_owned()];
         let mut proof =
             MKProof::from_leaves(leaves_to_verify).expect("MKProof generation should not fail");
-        proof.inner_root = leaves[10].to_owned();
+        proof.inner_root = Arc::new(leaves[10].to_owned());
         proof.verify().expect_err("The MKProof should be invalid");
     }
 
+    #[test]
+    fn test_should_list_leaves() {
```

Comment:
Same comment as the one on the `MKMap` test: by hand values should be preferred for those kind of tests.

Reply:
I don't understand your point here, what would be the difference if I wrote leaves by hand here?

Reply:
Done :heavy_check_mark: